src/pages/contact.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from 'react';
+import IPage from '../interfaces/IPage';
+import logging from '../config/logging';
+import { RouteComponentProps, withRouter } from 'react-router-dom';
+import { Form, Button } from "react-bootstrap"
+import firebase from "../db/firebase";
+
+const ContactPage: React.FunctionComponent<IPage & RouteComponentProps<any>> = props => {
+
+  useEffect(() => {
+    logging.info(`Loading ${props.name}`);
+  }, [props.name])
+
+  type formFieldsType = {
+    authorName: string,
+    email: string,
+    subject: string,
+    message: string,
+  }
+
+  function sendMessageToDb(formFields:formFieldsType): void {
+    firebase.firestore().collection("contactus").doc()
+    .set({
+      authorName: formFields.authorName,
+      email: formFields.email,
+      subject: formFields.subject,
+      message: formFields.message,
+      timestamp: new Date().toUTCString()
+    })
+    .then(() => {
+      console.log('Message successfully submitted!');
+    })
+    .catch((error) => {
+      console.error("Error sending message: ", error);
```

Comment:
You might want to let this error bubble up and catch it in `onformSubmit`. Then you could avoid resetting the fields if there was an error.

Reply:
Though I guess if you do that, you're also signing up for blocking input on the form while waiting for the firestore update to resolve. You might want to do that anyway, to avoid having duplicate sets in flight. Also, you might want to check if the fields are empty to prevent people from submitting a bunch of empty submissions.

Reply:
Good point, addressed in 6e6b3fc
Re checking the empty fields - all the fields are required so we're all set with that.